src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs

```diff
@@ -54,6 +57,7 @@ protected override Task InitializeAsync(CancellationToken cancellationToken, IPr
             _factory = new MigrateXprojProjectFactory(new ProcessRunner(), new FileSystem());
             _factory.SetSite(this);
             RegisterProjectFactory(_factory);
+            RegisterEditorFactory(new LoadedProjectFileEditorFactory(this));
```

Comment:
It hasn't crashed anything yet, but probably not. I'll run a switch to the ui thread.